config/jobs/kubernetes/test-infra/test-infra-trusted.yaml

```diff
@@ -52,6 +53,7 @@ postsubmits:
 
   - name: post-test-infra-push-prow
     cluster: test-infra-trusted
+    run_if_changed: 'prow/'
```

Comment:
This isn't technically sufficient since Prow depends on test-infra source code outside `prow/`, for example https://github.com/kubernetes/test-infra/blob/4471f77e877bfa421c46c7a5e93c7b838dedf2c3/prow/cmd/tide/main.go#L42

I'm not sure if we care.